### PR TITLE
Add segmentation model based on vit16 architecture.

### PR DESCRIPTION
**Pull request by ekellbuch**
Include segmentation model based on vit16+ architecture (https://arxiv.org/abs/2105.05633) following internal implementation in https://github.com/google-research/scenic.

### CODE REVIEW

File: uncertainty_baselines/models/segmenter.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        use_bias=self.use_bias,
+        precision=self.precision)(  # pytype: disable=wrong-arg-types
+            x)
+    output = nn.Dropout(
+        rate=self.dropout_rate)(
+            output, deterministic=deterministic)
+    return output
+
+
+class Encoder1DBlock(nn.Module):
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter.py:114]: Are these modules the same as in vit.py? If so, recommend importing them directly.

File: uncertainty_baselines/models/segmenter.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+                mlp_dim=self.mlp_dim,
+                num_layers=self.num_layers,
+                num_heads=self.num_heads,
+                dropout_rate=self.dropout_rate,
+                attention_dropout_rate=self.attention_dropout_rate,
+                )(x, train=train)
+
+    out['transformed'] = x
+
+    """
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter.py:311]: delete comment

**Review by dustinvtran — CHANGES_REQUESTED** [review]
Looks great! Subject to some minor changes.

File: uncertainty_baselines/models/segmenter.py (hunk 1 above)
Comment (ekellbuch) [reply uncertainty_baselines/models/segmenter.py:114]: the encoder in segmenter include an stochastic depth mask which consistently improves the performance of the model. (see Table 2 in https://arxiv.org/pdf/2105.05633.pdf) Would it be best to include a flag for stochastic depth in the vit model or keep the segmenter model as a stand alone implementation?

**Comment by ekellbuch** [comment]
@googlebot I signed it!

**Comment by ekellbuch** [comment]
@googlebot I fixed it.

File: uncertainty_baselines/models/segmenter.py (hunk 1 above)
Comment (dustinvtran) [reply uncertainty_baselines/models/segmenter.py:114]: Keeping it standalone here sounds good to me.

**Review by dustinvtran — APPROVED** [review]

**Comment by dustinvtran** [comment]
Merged in https://github.com/google/uncertainty-baselines/commit/fe95e35aa8d5c0debb1e0a64925e3e8384c105f2.
